geozero/src/wkb/wkb_writer.rs

```diff
@@ -292,114 +410,157 @@ impl<W: Write> FeatureProcessor for WkbWriter<'_, W> {}
 #[cfg(test)]
 mod test {
     use super::*;
-    use crate::wkb::{process_ewkb_geom, process_gpkg_geom};
+    use crate::wkb::process_wkb_type_geom;
     use crate::ToWkb;
 
-    fn ewkb_roundtrip(ewkb_str: &str, with_z: bool, srid: Option<i32>) -> bool {
+    fn roundtrip(
+        dialect: WkbDialect,
+        ewkb_str: &str,
+        dims: CoordDimensions,
+        srid: Option<i32>,
+        envelope: Vec<f64>,
+    ) {
         let wkb_in = hex::decode(ewkb_str).unwrap();
         let mut wkb_out: Vec<u8> = Vec::new();
-        let mut writer = WkbWriter::new(&mut wkb_out, WkbDialect::Ewkb);
-        writer.dims.z = with_z;
+        let mut writer = WkbWriter::new(&mut wkb_out, dialect);
+        writer.dims = dims;
         writer.srid = srid;
-        assert!(process_ewkb_geom(&mut wkb_in.as_slice(), &mut writer).is_ok());
-        let ok = wkb_out == wkb_in;
-        if !ok {
-            dbg!(hex::encode(&wkb_out));
-        }
-        ok
+        writer.envelope = envelope;
+        assert!(process_wkb_type_geom(&mut wkb_in.as_slice(), &mut writer, dialect).is_ok());
+
+        assert_eq!(hex::encode(wkb_in), hex::encode(wkb_out));
     }
 
     #[test]
     fn ewkb_geometries() {
         // SELECT 'POINT(10 -20)'::geometry
-        assert!(ewkb_roundtrip(
+        roundtrip(
```

Comment:
all these unit test changes - probably not worth changing - just keep the helper method `ewkb_roundtrip`

Reply:
Then we'd have four functions `ewkb_roundtrip`, `gpkg_roundtrip`, `spatialite_roundrip` and `mysql_roundtrip` with identical code..?

Reply:
ah, i see, ok, nvm, lets modify the tests